goal_src/engine/geometry/vol.gc

```diff
@@ -5,3 +5,272 @@
+                      (+! (-> arg1 length) 2)
+                      (+! (-> obj point-count) 2)
+                      )
+                    )
+                  (vector+! s1-0 s1-0 sv-144)
+                  (vector+! s1-0 s1-0 sv-224)
+                  (+! s0-0 2)
+                  )
+                )
+              )
+            )
+          )
+        (label cfg-42)
+        (set! sv-240 (+ sv-240 1))
+        )
+      (when (nonzero? s0-0)
+        (vector-float*! s1-0 s1-0 (/ 1.0 (the float s0-0)))
+        (cond
+          ((< (-> arg2 allocated-length) (+ (-> arg2 length) 2))
+           (format 0 "ERROR <AG>: vol-control #x~X out of volume normals~%" obj)
+           )
+          (else
+            (set! (-> arg2 data (-> arg2 length) quad) (-> s1-0 quad))
+            (set! (-> arg2 data (+ (-> arg2 length) 1) quad) (-> s2-0 s3-0 quad))
+            (+! (-> arg2 length) 2)
+            (let ((v1-112 (+ (-> obj normal-count) 2)))
+              (set! (-> obj normal-count) v1-112)
+              v1-112
+              )
+            )
+          )
+        )
+      )
+    )
+  #f
+  )
+
+(defmethod debug-draw plane-volume ((obj plane-volume))
+  (let* ((v1-0 (-> obj volume-type))
+         (s5-0 (cond
+                 ((= v1-0 'vol)
+                  (the-as uint #x8000c000)
+                  )
+                 ((= v1-0 'pvol)
+                  (the-as uint #x80808080)
+                  )
+                 (else
+                   (the-as uint #x800000c0)
+                   )
+                 )
+               )
+         (s4-0 (-> obj first-point))
+         )
+    (dotimes (s3-0 (/ (-> obj point-count) 2))
+      (add-debug-line
+        #t
+        (bucket-id debug-draw1)
+        (the-as vector s4-0)
+        (the-as vector (&-> s4-0 4))
+        (the-as rgba s5-0)
+        #f
+        (the-as rgba -1)
+        )
+      (set! s4-0 (&-> s4-0 8))
+      )
+    )
+  0
+  (none)
+  )
+
+(defmethod dummy-11 plane-volume ((obj plane-volume) (arg0 vector) (arg1 float))
+  (dotimes (v1-0 (-> obj num-planes))
+    (if (< arg1 (- (vector-dot arg0 (the-as vector (-> obj plane v1-0))) (the-as float (-> obj plane v1-0 w))))
+        (return #f)
+        )
+    )
+  #t
+  )
+
+(defmethod dummy-9 vol-control ((obj vol-control))
+  (with-pp
+    (let ((a0-1 obj))
+      (when (and (and *display-vol-marks* (logtest? (-> a0-1 flags) 1)) (logtest? (-> obj flags) 2))
+        (when (zero? (-> obj debug-point))
+          (let ((s5-0 pp))
+            (let ((s6-1 (-> obj process)))
+              )
+            (let ((s4-0 0))
+              (dotimes (v1-8 (-> obj pos-vol-count))
+                (+! s4-0 (-> obj pos-vol v1-8 num-planes))
+                )
+              (dotimes (v1-11 (-> obj neg-vol-count))
+                (+! s4-0 (-> obj neg-vol v1-11 num-planes))
+                )
+              (set! (-> obj debug-point) (new 'debug 'vector-array (* 10 s4-0)))
```

Comment:
probably this used to be `new 'process`, and that's why they changed the process pointer.  But it doesn't matter - allocating on the debug heap doesn't check the current process in any way...